core/state/snapshot/utils.go

```diff
@@ -72,16 +74,84 @@ func checkDanglingDiskStorage(chaindb ethdb.KeyValueStore) error {
 // checkDanglingMemStorage checks if there is any 'dangling' storage in the journalled
 // snapshot difflayers.
 func checkDanglingMemStorage(db ethdb.KeyValueStore) error {
-	log.Info("Checking dangling journalled storage")
 	start := time.Now()
-	iterateJournal(db, func(pRoot, root common.Hash, destructs map[common.Hash]struct{}, accounts map[common.Hash][]byte, storage map[common.Hash]map[common.Hash][]byte) error {
+	log.Info("Checking dangling journalled storage")
+	err := iterateJournal(db, func(pRoot, root common.Hash, destructs map[common.Hash]struct{}, accounts map[common.Hash][]byte, storage map[common.Hash]map[common.Hash][]byte) error {
 		for accHash := range storage {
 			if _, ok := accounts[accHash]; !ok {
 				log.Error("Dangling storage - missing account", "account", fmt.Sprintf("%#x", accHash), "root", root)
 			}
 		}
 		return nil
 	})
+	if !errors.Is(err, errNoJournal) {
+		log.Info("Failed to resolve snapshot journal", "err", err)
+		return err
+	}
```

Comment:
This can't be right. But changing it to `if errors.Is` won't make it correct either, I'm not quite sure why the distinction between errors is needed here?

Reply:
I'll merge it and then fix it

Reply:
About this -- is there any reason to not just do 
```diff 
diff --git a/core/state/snapshot/utils.go b/core/state/snapshot/utils.go
index 505ad15a99..b88ea05484 100644
--- a/core/state/snapshot/utils.go
+++ b/core/state/snapshot/utils.go
@@ -84,7 +84,7 @@ func checkDanglingMemStorage(db ethdb.KeyValueStore) error {
 		}
 		return nil
 	})
-	if !errors.Is(err, errNoJournal) {
+	if err != nil {
 		log.Info("Failed to resolve snapshot journal", "err", err)
 		return err
 	}

```